### DISCUSSION TITLE
only 2-3 background jobs were executed per night

### DISCUSSION THREAD

**Discussion by 42ske** [nextcloud/recognize#1135] [Q&A] [answered]
If I understood it correctly a background job should be executed every five minutes. On my installation since a few days (after a recrawl) background jobs only seams to be executed by night. And only two or three are executed. At the moment /settings/admin/recognize states:

`Face recognition: 25708 Queued files, Last classification: 4 days ago, Scheduled background jobs: 1, Last background job execution: 14 hours ago`

(two days ago it was about 26355 queed files and 7 scheduled background jobs)

`Face clustering: 8 faces left to cluster, Last clustering run: 4 days ago, Scheduled background jobs: 0, A minimum of 120 faces per user is necessary for clustering to kick in`

`Object recognition: 25832 Queued files, Last classification: 16 hours ago, Scheduled background jobs: 1, Last background job execution: 14 hours ago`

`Landmark recognition: 12 Queued files, Last classification: 16 hours ago, Scheduled background jobs: 1, Last background job execution: 14 hours ago`

Any ideas what I can do to make the face recognition a bit faster?

I'm using Recognize 6.1.1 on Nextcloud 28.0.3 (can't update myself, it’s a managed nextcloud)
Linux 5.15.0-101-generic x86_64
PHP 8.2.18
mysql 10.5.24

**Comment by marcelklehr** [comment, marked as answer]
If a maintenance window is set for your nextcloud that means recognize will only run during the maintenance window. You can increase the batch size for the classifiers to run through more photos per job.

**Reply by 42ske** [reply]
thanks. That helps to understand what's happening.

**Reply by luxzg** [reply]
Indeed thanks for this explanation. It was weird that jobs were running for several hours then abruptly stopped.
Since my instance is just home server, I've decided to set the window value to "100" as is recommended to allow jobs running at any time, and still removing the warning message from the Web UI. Jobs resumed again in a matter of minutes after changing the value.

```sudo -u www-data php /var/www/nextcloud/occ config:system:set maintenance_window_start --type=integer --value=100```
